src/error.rs

```diff
@@ -2,12 +2,13 @@
 //!
 
 use alloc::string::String;
+#[cfg(not(feature = "std"))]
+use core::error;
 use core::fmt;
 use core::num::TryFromIntError;
 use core::result;
 #[cfg(feature = "std")]
 use std::{error, io};
```

Comment:
I think we can delete this entire line and cfg by writing full import on the io variant of the error, eg `IO(std::io::Error)`